### ISSUE TITLE
help with install

### ISSUE THREAD

**Issue by mschendel** [jonescompneurolab/hnn#252]
Hello.
We had IT set up a VM which I thought had everything we needed to run HNN, but when I try python3 hnn.py I get the following errors, see below. 
I try to use the curl command, and it doesn't seem to do anything.
I have some experience with python, and I've used conda to manage some other python environments, but from the instructions on the ubuntu install, I don't really see what step to take next to check my system & troubleshoot.  What advice can you give?  
Thank you
Megan

System details (Please let me know what other info to check)
> lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic

python3 hnn.py
Traceback (most recent call last):
  File "hnn.py", line 10, in <module>
    from hnn_qt5 import *
  File "/opt/hnn_source_code/hnn_qt5.py", line 29, in <module>
    import nlopt
  File "/export/research/analysis/human/jstephen/shared/programs/python/anaconda3/lib/python3.5/site-packages/nlopt.py", line 35, in <module>
    _nlopt = swig_import_helper()
  File "/export/research/analysis/human/jstephen/shared/programs/python/anaconda3/lib/python3.5/site-packages/nlopt.py", line 31, in swig_import_helper
    _mod = imp.load_module('_nlopt', fp, pathname, description)
  File "/export/research/analysis/human/jstephen/shared/programs/python/anaconda3/lib/python3.5/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/export/research/analysis/human/jstephen/shared/programs/python/anaconda3/lib/python3.5/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: libnlopt.so.0: cannot open shared object file: No such file or directory

curl --remote-name https://raw.githubusercontent.com/jonescompneurolab/hnn/master/installer/ubuntu/hnn-ubuntu.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (77) error setting certificate verify locations:
  CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none

**Comment by blakecaldwell** [comment]
Hi. I see you are trying the installation without running the provided script. It looks like something happened with the `nlopt` installation. Could you post the output from the install script?
```
bash hnn-ubuntu.sh
```

If you want to use Anaconda instead of the script, that is a fine option, but our ability to troubleshoot may be limited. You can follow along with the script substituting `conda` where appropriate. However, I see that you are using Python 3.5. I would recommend Python 3.6 or 3.7. Note that you might run still run into problems similar to aboce above with `nlopt` because `conda` will install this module differently than `pip`.

**Comment by mschendel** [comment]
Hi
Thanks for your response.
I think because the curl isn't working, I don't have everything I need.  Actually, the hnn-ubuntu.sh is not there.  
I would prefer to follow install instructions as provided, just mentioning the conda to give you some general feel for my level of experience (which is pretty limited!)
I am trying to pick up what IT did, and I'm not sure what part of the install they were able to complete.  I have a the hnn_source_code directory, but I don't have the hnn-ubuntu.sh file. See below, you can see the sh files I have.

Any suggestion for why the curl command isn't working? 
Thanks a bunch,
Megan

nss:/opt/hnn_source_code> ls ./*/*sh
./scripts/docker_functions.sh*  ./scripts/setup-travis-mac.sh*      ./scripts/utils.sh*
./scripts/run-travis-wsl.sh*    ./scripts/setup-travis-windows.sh*

**Comment by blakecaldwell** [comment]
Sure, we can help. I don't know about the `curl` issue. Since it's only purpose is to download the script, maybe try `wget` instead.

Truthfully, it might be most informative to use the script as a guide, following the logic in it, and copying lines one-by-one.  If you click on the link above, you can at least see the script in your browser.

The instructions for Mac may help a little, though beware as many steps are not necessary for linux (e.g. creating conda activate.d files).

You might have an outdated copy of the source code? Consider a new `git clone` from this repository.

**Comment by mschendel** [comment]
Thanks for your help!
I was able to get the script via wget, thanks for that suggestion.  I ran the script, but now that I'm a bit more familiar, it appears the script was already run.  I am copying below the text from the first install log, run not by me.  And beneath that, the text of the 2nd install log which claims the installation was successful.  Despite this log, I was getting the same error when I tried to run hnn.py

However, I was able to fix the error just by using apt-get to install libnlopt0. I'm not sure if this is something missing from my VM due to oversight, or whether it should be in the ubuntu_install.sh ?  
Anyway, Thanks very much for your help!  Looking forward to trying the software!
Megan

Here is the text from the first install log:
> vi ubuntu_install.log
Requirement already satisfied: matplotlib in /root/.local/lib/python3.6/site-packages (3.3.2)
Requirement already satisfied: PyOpenGL in /root/.local/lib/python3.6/site-packages (3.1.5)
Requirement already satisfied: pyqt5 in /root/.local/lib/python3.6/site-packages (5.15.1)
Requirement already satisfied: pyqtgraph in /root/.local/lib/python3.6/site-packages (0.11.0)
Requirement already satisfied: scipy in /root/.local/lib/python3.6/site-packages (1.5.3)
Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (1.19.2)
Requirement already satisfied: nlopt in /root/.local/lib/python3.6/site-packages (2.6.1)
Requirement already satisfied: psutil in /root/.local/lib/python3.6/site-packages (5.7.2)
Requirement already satisfied: kiwisolver>=1.0.1 in /root/.local/lib/python3.6/site-packages (from matplotlib) (1.2.0)
Requirement already satisfied: certifi>=2020.06.20 in /root/.local/lib/python3.6/site-packages (from matplotlib) (2020.6.20)
Requirement already satisfied: cycler>=0.10 in /root/.local/lib/python3.6/site-packages (from matplotlib) (0.10.0)
Requirement already satisfied: python-dateutil>=2.1 in /root/.local/lib/python3.6/site-packages (from matplotlib) (2.8.1)
Requirement already satisfied: pillow>=6.2.0 in /root/.local/lib/python3.6/site-packages (from matplotlib) (8.0.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /root/.local/lib/python3.6/site-packages (from matplotlib) (2.4.7)
Requirement already satisfied: PyQt5-sip<13,>=12.8 in /root/.local/lib/python3.6/site-packages (from pyqt5) (12.8.1)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from cycler>=0.10->matplotlib) (1.11.0)
Downloading and extracting HNN...

and the text from the 2nd install log:
 vi ubuntu_install.log
HNN installation successful
Source code is at /opt/hnn_source_code
HNN installation successful
Source code is at /opt/hnn_source_code

Here is what is in the directory:
./                   feed.py          _layouts/           README.md
../                  fileio.py        lfp.py              requirements.txt
ac_manu_gamma.py     filt.py          LICENSE             res/
assets/              Gemfile          loadmodel_nrnui.py  run.py*
averaging.py         .gitignore       Makefile            scripts/
axes_create.py       gutils.py        mod/                seg3d.py
cartesian.py         hnn*             morphology.py       simdat.py
cell.py              hnn.cfg          netParams.py        specfn.py
cfg.py               hnn.desktop*     network.py          spikefn.py
clidefs.py           hnn_docker.sh*   nsgr.py             tests/
cli.py               hnn_nrnui.py     param/              .travis.yml
CODE_OF_CONDUCT.md   hnn.png          paramrw.py          ubuntu_install.log
_config.yml          hnn.py*          params_default.py   visdipole.py
conf.py              hnn_qt5.py       plotfn.py           vislfp.py
CONTRIBUTING.md      __init__.py      pmanu_gamma.py      visnet.py
ctune.py             init.py          ppsth.py            vispsd.py
currentfn.py         installer/       praster.py          visrast.py
data/                L2_basket.py     praw.py             visspec.py
DataViewGUI.py       L2_pyramidal.py  pspec.py            visvolt.py
dipolefn.py          L5_basket.py     ptest.py            x86_64/
example_analysis.py  L5_pyramidal.py  PT_example.py
